pkg/apis/core/types.go

```diff
@@ -5500,6 +5513,14 @@ type PodLogOptions struct {
 	// the actual log data coming from the real kubelet).
 	// +optional
 	InsecureSkipTLSVerifyBackend bool
+
+	// Specify which container log stream to return to the client.
+	// Acceptable values are "All", "Stdout" and "Stderr". If not specified, "All" is used, and both stdout and stderr
+	// are returned interleaved.
+	// Note that when "TailLines" is specified, "Stream" can only be set to nil or "All".
```

Comment:
Thanks for the great guidance!

Reply:
I agree, that a non-nil typeref to a string is fine here.  Clients are only ever writers of query params so the API behavior here is quite a bit simpler than for resource fields.

Reply:
Whether pointer or not, kubelet code still needs to test (for nil or "") because of the corner case where there are no params provided (and defaulting does not kick in).